Userland/DevTools/HackStudio/Editor.cpp

```diff
@@ -629,47 +629,51 @@ void Editor::set_cursor(const GUI::TextPosition& a_position)
 
 void Editor::set_syntax_highlighter_for(CodeDocument const& document)
 {
-    switch (document.language()) {
-    case Language::Cpp:
-        if (m_use_semantic_syntax_highlighting) {
-            set_syntax_highlighter(make<Cpp::SemanticSyntaxHighlighter>());
-            on_token_info_timer_tick();
-            m_tokens_info_timer->restart();
-        } else
-            set_syntax_highlighter(make<Cpp::SyntaxHighlighter>());
-
-        break;
-    case Language::CMake:
-        set_syntax_highlighter(make<CMake::SyntaxHighlighter>());
-        break;
-    case Language::CMakeCache:
-        set_syntax_highlighter(make<CMake::Cache::SyntaxHighlighter>());
-        break;
-    case Language::CSS:
-        set_syntax_highlighter(make<Web::CSS::SyntaxHighlighter>());
-        break;
-    case Language::GitCommit:
-        set_syntax_highlighter(make<GUI::GitCommitSyntaxHighlighter>());
-        break;
-    case Language::GML:
-        set_syntax_highlighter(make<GUI::GML::SyntaxHighlighter>());
-        break;
-    case Language::HTML:
-        set_syntax_highlighter(make<Web::HTML::SyntaxHighlighter>());
-        break;
-    case Language::JavaScript:
-        set_syntax_highlighter(make<JS::SyntaxHighlighter>());
-        break;
-    case Language::Ini:
-        set_syntax_highlighter(make<GUI::IniSyntaxHighlighter>());
-        break;
-    case Language::Shell:
-        set_syntax_highlighter(make<Shell::SyntaxHighlighter>());
-        break;
-    case Language::SQL:
-        set_syntax_highlighter(make<SQL::AST::SyntaxHighlighter>());
-        break;
-    default:
+    if (document.language().has_value()) {
```

Comment:
This would be a bit cleaner IMO with an early return (and make the diff easier to read):
```c++
if (!document.language().has_value()) {
    set_syntax_highlighter({});
    return;
}

switch (*document.language()) {
.
.
.
}
```